sql/get_package.sql

```diff
@@ -1,3 +1,20 @@
+with
+  retired_package_ids as (
+    -- A package is retired if its latest release is retired
+    select package_id from (
+      select
+        package_id
+        , retirement_reason
+      from
+        releases
+      group by
+        package_id
+      having
+        max(inserted_in_hex_at)
+    )
+    where
+      retirement_reason is not null
+  )
```

Comment:
this bit is a little repetitive

Reply:
This would be a good place to use a view! Let's do that instead. Once it has been declared it could be used like so:

```sql
select ... from non_retired_packages where ...
```

Does `get_package` need to be limited to unretired packages? Wouldn't that cause importing to fail if it is un-retired?

Reply:
It doesn't need to be limited, though it seems `get_package` is not used at all (`index.get_package` is not called anywhere except for tests)

```
> git grep -w get_package
src/packages/generated/sql.gleam:pub fn get_package(
src/packages/index.gleam:pub fn get_package(db: Connection, id: Int) -> Result(Option(Package), Error) {
src/packages/index.gleam:  use returned <- result.then(sql.get_package(db.inner, params, decode_package))
test/packages/index_test.gleam:  let assert Ok(Some(package)) = index.get_package(db, id)
test/packages/index_test.gleam:  let assert Ok(None) = index.get_package(db, id + 1)
```

Reply:
Yup, it is currently the case as this diff is outdated. So I'm going to click resolve on this